python/lsst/ts/electrometer/csc.py

```diff
@@ -27,86 +80,180 @@ def detailed_state(self, new_state):
         self.report_detailed_state()
 
     def report_detailed_state(self):
+        """Report the new detailed state."""
         self.evt_detailedState.set_put(detailedState=self.detailed_state)
 
     async def configure(self, config):
-        self.model.configure(config)
+        """Configure the Electrometer CSC.
 
-    async def implement_simulation_mode(self, mode):
-        self.model.implement_simulation_mode(simulation_mode=mode)
+        Parameters
+        ----------
+        config : types.SimpleNamespace
+            The parsed yaml object.
+        """
+        self.controller.configure(config)
 
     async def end_enable(self, id_data):
-        self.model.connect()
+        """Connect the electrometer and configure to the settings specified.
+
+        Parameters
+        ----------
+        id_data : data
+            The data for the hook.
+        """
+        await self.controller.connect()
+        self.evt_measureType.set_put(mode=self.controller.mode.value)
+        self.evt_digitalFilterChange.set_put(activateFilter=self.controller.filter_active,
+                                             activateAverageFilter=self.controller.avg_filter_active,
+                                             activateMedianFilter=self.controller.median_filter_active)
+        self.evt_integrationTime.set_put(intTime=self.controller.integration_time)
+        self.evt_measureRange.set_put(rangeValue=self.controller.range)
         self.detailed_state = Electrometer.DetailedState.NOTREADINGSTATE
 
-    async def begin_disable(self, id_data):
-        self.model.disconnect()
+    async def end_disable(self, id_data):
+        """Disconnect the electrometer.
+
+        Parameters
+        ----------
+        id_data : data
+            The data for the hook.
+        """
+        self.controller.disconnect()
 
     async def do_performZeroCalib(self, id_data):
+        """Perform zero calibration.
+
+        Parameters
+        ----------
+        id_data : data
+            The data for the command.
+        """
         self.assert_enabled("performZeroCalib")
         self.assert_substate(substates=[Electrometer.DetailedState.NOTREADINGSTATE],
                              action="performZeroCalib")
         self.detailed_state = Electrometer.DetailedState.CONFIGURINGSTATE
-        self.model.perform_zero_calib()
+        await self.controller.perform_zero_calibration()
         self.detailed_state = Electrometer.DetailedState.NOTREADINGSTATE
 
     async def do_setDigitalFilter(self, id_data):
+        """Set the digital filter(s).
+
+        Parameters
+        ----------
+        id_data : data
+            The data for the command.
+        """
         self.assert_enabled("setDigitalFilter")
```

Comment:
Really, I didn't know that.

Reply:
That's a relief. I'm not sure who I've told what. Feel free to leave the strings arguments or remove them. I suggest writing new code without the argument specified because it avoids copy/paste errors.